hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/UDTAttributesTable.java

```diff
@@ -0,0 +1,188 @@
+                                ? Integer.MAX_VALUE
+                                : null, // character_octet_length
+                        null, // character_set_catalog
+                        null, // character_set_schema
+                        null, // character_set_name
+                        null, // collation_catalog
+                        null, // collation_schema
+                        null, // collation_name
+                        field.getQueryDataType().getTypeFamily().isNumeric()
+                                ? getNumericTypePrecision(field.getQueryDataType())
+                                : null, // numeric_precision
+                        field.getQueryDataType().getTypeFamily().isNumeric()
+                                ? 2
+                                : null, // numeric_precision_radix
+                        field.getQueryDataType().getTypeFamily().isNumericInteger()
+                                ? 0
+                                : null, // numeric_scale
+                        field.getQueryDataType().getTypeFamily().isTemporal()
+                                ? getTemporalTypePrecision(field.getQueryDataType())
+                                : null, // datetime_precision
+                        null, // interval_type
+                        null, // interval_precision
+                        field.getQueryDataType().isCustomType()
+                                ? catalog()
+                                : null, // attribute_udt_catalog
+                        field.getQueryDataType().isCustomType()
+                                ? schema
+                                : null, // attribute_udt_schema
+                        field.getQueryDataType().isCustomType()
+                                ? field.getQueryDataType().getObjectTypeName()
+                                : null, // attribute_udt_name
+                        null, // scope_catalog
+                        null, // scope_schema
+                        null, // scope_name
+                        null, // maximum_cardinality
+                        null, // dtd_identifier
+                        null  // is_derived_reference_attribute
+                });
+            }
+        }
+
+        return rows;
+    }
+
+    private String toSqlDataTypeString(final QueryDataType dataType) {
+        if (dataType.isCustomType()) {
+            return "USER-DEFINED";
+        }
+        return HazelcastTypeUtils.toCalciteType(dataType).getName();
+    }
+
+    @SuppressWarnings("checkstyle:MagicNumber")
+    private Integer getTemporalTypePrecision(final QueryDataType dataType) {
+        switch (dataType.getTypeFamily()) {
+            case TIME:
+            case TIMESTAMP:
+            case TIMESTAMP_WITH_TIME_ZONE:
+                // number of digits after the dot when specifying seconds - 9 for nanoseconds
+                return 9;
```

Comment:
This bit was confusing - the value here is number of digits (according to Postgres docs) after the dot, Postgres provides 6 here for both TIMESTAMP and TIMESTAMP WITH TIME ZONE here, which means microseconds, but I think we return OffsetDateTime which allows querying for Nanoseconds? Not sure what's best.